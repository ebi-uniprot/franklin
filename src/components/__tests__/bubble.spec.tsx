import { render } from '@testing-library/react';

import Bubble from '../bubble';

describe('Long number component', () => {
  test('should render with defaults', () => {
    const { asFragment } = render(<Bubble>{10}</Bubble>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render bigger than 100', () => {
    const { asFragment } = render(<Bubble>{101}</Bubble>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render different size', () => {
    const { asFragment } = render(<Bubble size="large">{101}</Bubble>);
    expect(asFragment()).toMatchSnapshot();
  });
});
