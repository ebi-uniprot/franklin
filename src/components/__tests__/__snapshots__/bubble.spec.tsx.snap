// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Long number component should render bigger than 100 1`] = `
<DocumentFragment>
  <span
    class="bubble--medium"
  >
    99+
  </span>
</DocumentFragment>
`;

exports[`Long number component should render different size 1`] = `
<DocumentFragment>
  <span
    class="bubble--large"
  >
    99+
  </span>
</DocumentFragment>
`;

exports[`Long number component should render with defaults 1`] = `
<DocumentFragment>
  <span
    class="bubble--medium"
  >
    10
  </span>
</DocumentFragment>
`;

exports[`Long number component should render without children 1`] = `
<DocumentFragment>
  <span
    class="bubble--medium"
  />
</DocumentFragment>
`;
